chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -1374,81 +1347,42 @@ impl PeerManagerActor {
         // Check if this is a blacklisted peer.
         if peer_info.addr.as_ref().map_or(true, |addr| self.peer_store.is_blacklisted(addr)) {
             debug!(target: "network", peer_info = ?peer_info, "Dropping connection from blacklisted peer or unknown address");
-            return RegisterPeerResponse::Reject;
+            return RegisterPeerResponse::Reject(RegisterPeerError::Blacklisted);
         }
 
         if self.peer_store.is_banned(&peer_info.id) {
             debug!(target: "network", id = ?peer_info.id, "Dropping connection from banned peer");
-            return RegisterPeerResponse::Reject;
-        }
-
-        let tier2 = self.state.tier2.load();
-        if msg.connection.peer_type == PeerType::Inbound
-            && !self.is_inbound_allowed()
-            && !self.is_peer_whitelisted(&peer_info)
-        {
-            // TODO(1896): Gracefully drop inbound connection for other peer.
-            debug!(target: "network",
-                tier2 = tier2.ready.len(), outgoing_peers = tier2.outbound_handshakes.len(),
-                max_num_peers = self.max_num_peers,
-                "Inbound connection dropped (network at max capacity)."
-            );
-            return RegisterPeerResponse::Reject;
-        }
-
-        let other_edge_info = &msg.connection.partial_edge_info;
-        if other_edge_info.nonce == 0 {
```

Comment:
Did we lost this check?

Reply:
right, I'll move this check to verify_nonce.